### PR TITLE
buildversioninfo: store file link using relative path rather than absolute path

### PR DESCRIPTION
**Pull request by ansyral**
@herohua @vwxyzh @superyyrrzz @hellosnow @qinezh @chenkennt

### CODE REVIEW

**Comment by herohua** [comment]
:+1:

File: src/Microsoft.DocAsCode.Common/JsonUtility.cs [hunk 1]
```diff
@@ -20,24 +21,28 @@ public static class JsonUtility
...
 
-        public static void Serialize(TextWriter writer, object graph, Formatting formatting = Formatting.None, JsonSerializer serializer = null)
+        public static void Serialize(TextWriter writer, object graph, Formatting formatting = Formatting.None, JsonSerializer serializer = null, object additional = null)
         {
             var localSerializer = serializer ?? DefaultSerializer.Value;
             localSerializer.Formatting = formatting;
+            if (additional != null)
+            {
+                localSerializer.Context = new StreamingContext(StreamingContextStates.Other, additional);
+            }
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/JsonUtility.cs]: bad status:
Call this method twice, first time with some additional, second time without additional.

File: src/Microsoft.DocAsCode.Common/JsonUtility.cs [hunk 2]
```diff
@@ -20,24 +21,28 @@ public static class JsonUtility
                     return jsonSerializer;
                 });
 
-        public static void Serialize(TextWriter writer, object graph, Formatting formatting = Formatting.None, JsonSerializer serializer = null)
+        public static void Serialize(TextWriter writer, object graph, Formatting formatting = Formatting.None, JsonSerializer serializer = null, object additional = null)
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/JsonUtility.cs]: >, object additional = null [](start = 143, length = 26)

not binary compatible

**Comment by vwxyzh** [comment]
:clock1:

**Comment by hellosnow** [comment]
:shipit:

**Comment by ansyral** [comment]
#449

**Pull request merged**